src/plugin/morphologyHDF5.cpp

```diff
@@ -213,35 +213,53 @@ void MorphologyHDF5::_readPoints(unsigned int firstSectionOffset)
                 "Reading morphology file '" + _file->getName() + "': bad number of dimensions in 'points' dataspace"));
         }
         std::vector<std::vector<float>> vec(dims[0]);
-        // points.resize(dims[0]);
         dataset.read(vec);
-        bool noNeurites = (firstSectionOffset == -1);
-        for (unsigned int i = 0; i < vec.size(); ++i) {
+
+        if (firstSectionOffset < 0) {
+            firstSectionOffset = vec.size();
```

Comment:
For some reason I still feel leery about re-setting `firstSectionOffset` to the value which gives the same effect as the previous code.
I would suggest to limit this review to unquestionable things like fixing dtype, adding compilation warnings and `.reserve()` calls and see with Benoit about this refactoring.
Alternatively, we can merge it as it is and undo part of it afterwards.

Reply:
I'd rather have it like this and move the `if` clause outside the loop. I'll merge this now, and we can revert as needed.